recipes/harfbuzz/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class HarfbuzzConan(ConanFile):
+    name = "harfbuzz"
+    description = "HarfBuzz is an OpenType text shaping engine."
+    topics = ("conan", "harfbuzz", "opentype", "text", "engine")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://harfbuzz.org"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/*.patch"]
+    generators = "cmake"
+    short_paths = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_freetype": [True, False],
+        "with_icu": [True, False],
+        "with_glib": [True, False],
+        "with_gdi": [True, False],
+        "with_uniscribe": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_freetype": True,
+        "with_icu": False,
+        "with_glib": False,
+        "with_gdi": True,
+        "with_uniscribe": True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _cmake = None
+
+    def requirements(self):
+        if self.options.with_freetype:
+            self.requires("freetype/2.10.1")
+        if self.options.with_icu:
+            self.requires("icu/64.2")
+        if self.options.with_glib:
+            self.requires("glib/2.65.0")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        else:
+            del self.options.with_gdi
+            del self.options.with_uniscribe
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake_compiler_flags(self, cmake):
+        flags = []
+        compiler = str(self.settings.compiler)
+        if compiler in ("clang", "apple-clang"):
+            flags.append("-Wno-deprecated-declarations")
+        cmake.definitions["CMAKE_C_FLAGS"] = " ".join(flags)
+        cmake.definitions["CMAKE_CXX_FLAGS"] = cmake.definitions["CMAKE_C_FLAGS"]
+
+        return cmake
+
+    def _configure_cmake_macos(self, cmake):
+        if str(self.settings.os) in ["Macos", "iOS", "watchOS", "tvOS"]:
```

Comment:
```suggestion
        if tools.is_apple_os(self.settings.os):
```
https://docs.conan.io/en/latest/reference/tools.html#tools-is-apple-os